api/v1/swagger.yaml

```diff
@@ -6160,6 +6160,239 @@ paths:
+            Success, returns all Bots
+          schema:
+            type: array
+            items:
+              $ref: "#/definitions/BotResponse"
+        400:
+          $ref: "#/responses/400"
+        401:
+          $ref: "#/responses/401"
+        404:
+          $ref: "#/responses/404"
+        default:
+          $ref: "#/responses/genericError"
+
+    post:
+      security:
+        - jwt: []
+      summary: Create a new bot
+      tags: [ bots ]
+      description: Create a new bot with a name
+      operationId: postBots
+      parameters:
+        -
+          name: queryBody
+          in: body
+          required: true
+          schema:
+            type: object
+            description: Create bot properties
+            properties:
+              name:
+                type: string
+                maxLength: 254
+                description: Name of the bot
+              url:
+                type: string
+                description: Bot url
+              active:
+                type: boolean
+                description: If the bot is active
+              actions:
+                type: array
+                description: Array of objects that contain the name of actions a bot can take and an array of parameters needed for that action
+                items:
+                  type: object
+              data:
+                type: array
+                description: Array of objects that contain the name of data on a bot
+                items:
+                  type: object
+            required:
+              - name
+      responses:
+        201:
+          description: >-
+            Created bot
+          schema:
+            $ref: "#/definitions/BotResponse"
+        400:
+          $ref: "#/responses/400"
+        401:
+          $ref: "#/responses/401"
+        403:
+          $ref: "#/responses/403"
+        default:
+          $ref: "#/responses/genericError"
+
+# ---------------------------------------------------------------------------
+  /bots/{key}:
+    x-swagger-router-controller: bots
+    delete:
+      security:
+        - jwt: []
+      summary: Delete the specified bot
+      tags: [ bots ]
+      description: >-
+        Delete a specific bot
+      operationId: deleteBots
+      parameters:
+        -
+          name: key
+          in: path
+          description: >-
+            The id or name of the bot to delete.
+          required: true
+          type: string
+      responses:
+        200:
+          description: >-
+            Success, Deleted Bot
+          schema:
+            type: array
+            items:
+              $ref: "#/definitions/BotResponse"
+        400:
+          $ref: "#/responses/400"
+        401:
+          $ref: "#/responses/401"
+        404:
+          $ref: "#/responses/404"
+        default:
+          $ref: "#/responses/genericError"
+
+    get:
+      summary: Retrieve a bot
+      tags: [ bots ]
+      operationId: getBots
```

Comment:
This can only return ONE bot, so better name would be "getBot" rather than "getBots"